### PR TITLE
Update react-scripts to version 4.0 in client

### PR DESCRIPTION
**Pull request by sibille**

### CODE REVIEW

File: src/client/src/index.tsx [hunk 1]
```diff
@@ -23,7 +23,7 @@ const store = createStoreWithMiddleware(
 );
 
 const vscode = process.env.NODE_ENV === ENVIRONMENT.PRODUCTION ?
-  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
```
Comment (prupipho) [inline src/client/src/index.tsx:26]: Revert this change to what it was before. 
```suggestion
  // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
```

**Comment by prupipho** [comment]
Mind there´s now warnings that weren´t there before:
[See here](https://github.com/typescript-eslint/typescript-eslint/blob/v4.11.0/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md#configuring-in-a-mixed-jsts-codebase)
```
Type: warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types

✖ 176 problems (0 errors, 176 warnings)
```

**Comment by sibille** [comment]
> Mind there´s now warnings that weren´t there before:
> [See here](https://github.com/typescript-eslint/typescript-eslint/blob/v4.11.0/packages/eslint-plugin/docs/rules/explicit-module-boundary-types.md#configuring-in-a-mixed-jsts-codebase)
> 
> ```
> Type: warning  Missing return type on function  @typescript-eslint/explicit-module-boundary-types
> 
> ✖ 176 problems (0 errors, 176 warnings)
> ```

Created issue https://github.com/microsoft/WebTemplateStudio/issues/1706 to fix the warnings.

**Review by prupipho — APPROVED** [review]
